### PR TITLE
SPLICE-159 Supporting Hadoop Compliant File Systems

### PR DESCRIPTION
**Pull request by jleach4**
Supporting hadoop compliant filesystems.  Small change to file system abstraction.  Wrote a simple S3 test...

### CODE REVIEW

File: hbase_sql/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/S3IT.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                            "%d," +    // max bad records
+                            "'%s'," +  // bad record dir
+                            "null," +  // has one line records
+                            "null)",   // char set
+                    spliceSchemaWatcher.schemaName, "NATION2",
+                    LOCATION,
+                    0,
+                    BADDIR.getCanonicalPath()));
+            ps.execute();
+        ResultSet rs = methodWatcher.executeQuery("select * from nation2");
```
Comment (snowspeeder) [inline hbase_sql/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/S3IT.java]: might want to add order, not sure if order is always guarunteed

Comment (jleach4) [inline hbase_sql/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/S3IT.java]: The validation is unordered.

**Comment by genedavis** [comment]
+1

**Comment by jfilali** [comment]
+1 some methods miss java doc though

**Pull request merged**